### ISSUE TITLE
Positive mods for Zenith Tower

### ISSUE THREAD

**Issue by MitchOfTarcoola** [o5k/tetrio-issues#1391]
### Checklist

- [X] I've **checked both [open and recently closed issues](https://github.com/tetrio/issues/issues?q=is%3Aissue)** to ensure this feature is not already requested.
- [X] I am using the **newest version** of TETR.IO.
- [X] I am certain this is a **feature request**, not a bug.
- [X] I've **thought this feature through** properly.

### Is your feature request related to a problem? Please describe.

currently, there are no options to reduce the difficulty of a Zenith run.

Low level players normally top out before the 3rd or 4th floor, and therefore never get a chance to experience the sights and music of most of the tower while playing.

Additionally, sometimes I would like to do a Zenith run more casually, without ending up as overwhelmed by garbage or other mishaps.

### Describe the solution you'd like

My idea is to add a set of positive mods, to go alongside the existing challenge mods. This will allow lower-level players a bigger chance of reaching higher floors without impacting those who don't use the mods too much.

For balance reasons, you won't be able to earn highscore-type Zenith Tower badges using positive mods, and no positive mod will directly increase your offense.

A way this could be implemented into the UI visually could be having positive mods being negative versions of the Tarot cards used by the challenge mods. Most of the ideas below are inversions of existing challenges.

Here are some possible ideas I've brainstormed:

#### Zen mode
- You send no garbage and receive no garbage. (would-be sent lines still score as normal)
- Fatigue is changed to add 2, 3, 5, 5, 5 permanent lines (this forces a top-out at max fatigue).
- Zen mode boards are hidden when zen mode is not active, and vice versa.
- This mode is meant for those who just want to experience the music and views of the Tower in a stress-free environment.
- This mode could be given its own leaderboard, playing like a more endurance focused Blitz.

#### Jump Start
- Start at 50m (floor 2) with speed level 4.
- This allows players to jump past the usually awkward floor 1 mess.

#### Clean Garbage
- Garbage is noticeably cleaner (less likely to change column, remains clean for longer).
- The first garbage can never roll column 2 or 9.
- A more minor effect than some of the other ideas but allows for easier back-to-back stacking and downstacking.

#### Time Dilation
- Gravity is slower at all levels.
- Increases the time it takes for garbage to activate.
- Speed level decays slower.
- Fatigue starts 2 mins later.
- This one is a mix of smaller time-related buffs.

#### Mini Garbage
- Halves incoming garbage.
- Straightforward, will greatly reduce late game stress. Probably the strongest boost besides zen for lower-level players.

#### Bomb garbage
- Use Bomb garbage instead of standard garbage.
- Clears involving Bombs have their garbage capped at 4 lines worth, if the Bombs meant the clear removed more than 4 lines.
- Bombs are generally easier to clean than normal garbage, especially in the early game. The attack cap should hopefully ensure no massive spikes occur due to large clean bomb garbage stacks, which would cause unfair KOs

#### Spatial Shift
- Receive garbage as if you were at half your current altitude.
- This would allow consistent players to reach around twice as high, if you ignore fatigue.

#### No-Spin Purity
- Clearing garbage lines removes the bottom line of garbage.
- However, all spins are disabled.
- This is functionally the opposite of the All-spin challenge mod, giving a powerful tool for downstacking at the cost of spins.

#### Bigger Thrusters
- Doubles instant altitude and speed level increase from sent garbage and KOs
- Has no direct assistance to actual gameplay but can allow reaching higher floors and higher speed levels, particularly for more slow and steady players.

### Describe alternatives you've considered

The only current alternative for low and mid-level players to reach higher floors is to play Duos with a higher-level player. However, this usually results in the lower player being KO'd for most/all the time in the higher floors, forced to sit and watch. Finding a suitable Duo partner is its own challenge, especially for those without Supporter.

**Comment by ArcaderRonny12** [comment]
Why is no one else here? This would seriously benefit low-level players!

**Comment by MegaGamer22** [comment]
For real! I can't get past the offices with my skill cap!

**Comment by A278PlusPi** [comment]
Block stacking games like Tetris are meant to be very casual, although this game (TETR.IO) have a largely competitive component. With so many beginners (think of B ranks or lower) being stuck on lower floors, positive mods allow the player to experience the entire Zenith tower without being pressured by garbage. With that, I would like to add more possible mods that one can use:

**Infinite Hold/Movement**
This is the opposite of "No Hold". Players can move the tetromino as many times they want without the piece automatically being locked after 15 moves. In addition, players can switch between tetrominoes (via Hold) as many times they want.

**Zen Mode (alternative)**
An another version of the "Zen mode" except with a few differences:
- Players can have unlimited stock if they choose to (in reference to Tetris Effect: Connected)
- There will be no fatigue. The player will have 30 minutes or 5,000 meters (whichever happens first) to experience a single session of Zen Mode. When the altitude is reached or the time is up, the board will simply disappear and will show the results screen. There will be a countdown timer to accompany that instead of the elapsed timer.

**Time Dilation**
One buff I would like to add:
- Lock delay is significantly increased (3 seconds in Floor 1 to 1.2 seconds in Floor 10)

**Dynamic Duo**
- Both players will send and receive garbage as if there are playing solo. If a team receives garbage, the garbage will be received at a normal rate, alternating each player
- Shared Stock: A team will receive 5 lives at the start, with each player having 2 lives each. When one player dies, the surviving player can use the remaining shared lives. For each successful revival, a team will earn 2 extra lives.
- Both players can swap hold pieces if they wish to

**Beginner Mode**
- Not an opposite of "Expert Mode", but rather based on the SSL gimmick "Beginner Mode". The player will receive dominoes and trominoes, as well as I-pieces and T-pieces.

Hopefully this would be a thing for beginners. I have seen players that have less than 0.5 PPS, and 5 VS, so I hope to see positive mods eventually. Happy stacking!

**Comment by csq-curiosity** [comment]
While I mostly agree with positive mods existing, I think there needs a bit of caution when it comes to how it is applied. The primary concept of Quick Play is scaling the zenith tower while surviving the onslaught of attacks, which means that by its inception there is a component of challenge and accomplishment involved. Alleviating that challenge can be very helpful and - in my opinion - essential to balance out the low level players, but I do not believe that a zen-like counterpart for Quick Play - or any other mode that almost kills challenge - is necessary since it is basically a timed marathon mode that largely defeats the core of what makes Quick Play what it is, other than maybe having a number go up. So, here are my two cents about the mode discussed so far:

Zen mode, Zen mode (altenative): **Against**. If it _must_ exist, then I think it should go into the singleplayer category due to the multiplayer component being either nonexistent or existing in name only.

Jump Start: **Against**, and I think a separate thread on discussing the "awkward floor 1 mess" would be in order. There definitely is an issue with early game Quick Play because Floor 1 is a jumbled mess of players of all skill level so receiving garbage is usually a matter of luck. So yes, a cleanup with Floor 1 garbage distribution is necessary, but I don't think Jump Start is how you solve it. If positive mods were to exist, the nature of them would mean that all players, even high level ones, would have instant access to them. (Unless you were to argue for capping low level mods by TL rank or QP history but that is a whole another can of worms) This would mean that, even if positive mod records were to be unranked, technically nothing stops mid~high level players from using jump start mod and this would have an effect of moving the goalpost.

Clean Garbage: **For**. A straightforward opposite of Wheel of Fortune, and has a strong argument. Even with no mods, with terrible luck garbage can get messy as early as Floor 5, and clean garbage can be greatly helpful on lower levels. Although I do have to mention that since Beta 1.7.0:
> On floor 5 and below, clean Garbage wells will only spawn in columns 3-8.

so that part is already covered. With this mod on though I feel like having garbage _never_ (or not until F10) spawn on columns 2 or 9 would also be an option.

Time Dilation: **For**. Kind of an opposite of Tower. Honestly even with no mods the gravity kicks in a bit too fast, and I think this indirectly gatekeeps players that aren't used to 0 ARR to play for extended periods of time. However I am against A278PlusPi's suggestion on adding lock delay increase, and I think that buff should be a part of infinite hold/movement buff instead - read below.

Mini Garbage: **Split**. This one is a relatively straightforward opposite of Strength, but the implications of having less garbage is messier than meets the eye. This is because a clean 4 line garbage is easy to deal with an I piece resulting in a counterattack, but a 1~2 line garbage can be tricky to turn it into an attack. This would mean that a Mini Garbage buff may unironically end up being harder than having no mods; not immediately, but on late floors (I believe around Floor 7). This would mean that this buff in isolation can help on lower floors but would need some help from Clean Garbage on later floors.

Spatial Shift: **Split**. I think this counts as the opposite to Emperor. This would be an alternative to Mini Garbage that would be easier to implement.

Bomb garbage: **For**. I don't think there is a direct opposite for this one, but for making downstacking easier this would be a clever solution that is also reasonable. Also the garbage cap is a solid way of balancing attacks.

No-spin Purity: **For**. The opposite of Magician, and a genius one at that too. Since low level players are less likely to abuse spins to say the least, this would be one of the few modes that would be too penalizing for high level players to use while still helping low level players. I would find a different way of buffing though: Instead of removing the bottom line garbage, it would _re-align_ one line worth of garbage to be on the same column as the uppermost one, and if all garbage lines are already aligned then the next one is guaranteed to be on that column.

Bigger Thrusters: **Against**. Has a component that is opposite of Emperor, but I feel like this one trivializes challenge too directly without having players actually work for it. For me Time Dilation is where I would draw the line.

Infinite Hold/Movement: **For**, and if I had to choose _just one_ positive mod to be in the game this would be it. This would be a massive help for not just low level players, but also for mid level players practicing negative mods.

--------------------------------------------------------------

Reading this thread gave me a couple of ideas to work with. These ideas have the premise of being unranked, but I think some reversed mods can actually be negative mods instead of positive.

The Heirophant - Reduced garbage cap. "Let's work this out slowly, one step at a time."
If you tank garbage in Quick Play, up to 10 lines can enter the board at once. The Heirophant would reduce this garbage cap starting at 1 in Floor 1 up to 4 in Floor 10. (I am thinking 1 1 1 2 2 2 3 3 3 4) Further increase garbage cap by +2 for fatigue at the 9 and 11 minute mark. This idea stems from the old days when players were goofing around in free play lobbies with 100x garbage multiplier and 1 garbage cap.

Reversed Heirophant - The Delinquent "THESE MOUNTING OBLIGATIONS ARE BUT YOURS TO TAKE"
Garbage would be tanked in the segments it has arrived, in that order. (So if 1, 2 and 7 lines of garbage is queued, you would receive 1, 2 and 7 lines of garbage, in that order.) This would be closer to the garbage mechanics found in the likes of TE:C, except garbage cannot be cancelled; you must tank and clear the lines.

The World - Helping hand. "Discover what the seven pieces are able to do."
This would need some sort of a high-level bot implemented in the game's code. With this mod on, highlight the next piece that the bot thinks is optimal.

Reversed World - Passivity "YOUR FATE IS NOW ON THEIR SOULLESS HANDS"
Unlike normal World, receive 20 wound lines (the same unclearable lines you see in Magician/Warlock) if you do not follow the bot's instructions. However, ghost would be disabled, and all pieces would have randomized colors. For randomized colors, it will start as some pieces not matching the color scheme all the way down to a single mino having four different colors for each block.

**Comment by pasicist** [comment]
The Quick Play experience was designed so that lower level players can experience the higher floors as they get better. Not only would this system undermine the experience of getting to a new floor legitimately, it would also undermine the achievement of getting to F10. Simply put, not everyone will have the privelege of getting to the higher floors, and I see no problem with it. (seriously you can just listen to the songs on youtube)
